src/GraphQL.MicrosoftDI/ResolverBuilder.cs

```diff
@@ -0,0 +1,313 @@
+using System;
+using System.Threading.Tasks;
+using GraphQL.Builders;
+using GraphQL.Utilities;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.MicrosoftDI
+{
+    /// <summary>
+    /// A builder for field resolvers with no extra service types.
+    /// </summary>
+    public class ResolverBuilder<TSourceType, TReturnType>
+    {
+        private readonly FieldBuilder<TSourceType, TReturnType> _builder;
+        private bool _scoped;
+
+        /// <summary>
+        /// Initializes a new instance with the specified properties.
+        /// </summary>
+        public ResolverBuilder(FieldBuilder<TSourceType, TReturnType> builder, bool scoped)
+        {
+            _builder = builder;
+            _scoped = scoped;
+        }
+
+        /// <summary>
+        /// Specifies a type that is to be resolved via dependency injection during the resolver's execution.
+        /// </summary>
+        public ResolverBuilder<TSourceType, TReturnType, T1> WithType<T1>()
+            => new ResolverBuilder<TSourceType, TReturnType, T1>(_builder, _scoped);
+
+        /// <summary>
+        /// Specifies that the resolver should run within its own dependency injection scope.
+        /// </summary>
+        /// <returns></returns>
+        public ResolverBuilder<TSourceType, TReturnType> WithScope()
+        {
+            _scoped = true;
+            return this;
+        }
+
+        /// <summary>
+        /// Specifies the delegate to execute when the field is being resolved.
+        /// </summary>
+        public FieldBuilder<TSourceType, TReturnType> Resolve(Func<IResolveFieldContext<TSourceType>, TReturnType> resolver)
+            => _scoped ? _builder.ResolveScoped(resolver) : _builder.Resolve(resolver);
+
+        /// <inheritdoc cref="Resolve(Func{IResolveFieldContext{TSourceType}, TReturnType})"/>
+        public FieldBuilder<TSourceType, TReturnType> ResolveAsync(Func<IResolveFieldContext<TSourceType>, Task<TReturnType>> resolver)
+            => _scoped ? _builder.ResolveScopedAsync(resolver) : _builder.ResolveAsync(resolver);
+    }
+
+    /// <summary>
+    /// A builder for field resolvers with 1 extra service type.
+    /// </summary>
+    public class ResolverBuilder<TSourceType, TReturnType, T1>
+    {
+        private readonly FieldBuilder<TSourceType, TReturnType> _builder;
+        private bool _scoped;
+
+        /// <inheritdoc cref="ResolverBuilder{TSourceType, TReturnType}.ResolverBuilder(FieldBuilder{TSourceType, TReturnType}, bool)"/>
+        public ResolverBuilder(FieldBuilder<TSourceType, TReturnType> builder, bool scoped)
+        {
+            _builder = builder;
+            _scoped = scoped;
+        }
+
+        /// <inheritdoc cref="ResolverBuilder{TSourceType, TReturnType}.WithType{T1}"/>
+        public ResolverBuilder<TSourceType, TReturnType, T1, T2> WithType<T2>()
```

Comment:
Did you think about "short circuit" methods like `WithType<T1, T2, T3>` ? I think that in this form they will be called much more often than one by one, although the API of one by one calls can be called "fundamental".

Reply:
No, but that's a good idea.  I'll add it.